src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -47,7 +45,19 @@ class SyncController(
   override def receive: Receive = idle
 
   def idle: Receive = handlePeerUpdates orElse {
-    case StartFastSync =>
+
+    case StartSync =>
+      (appStateStorage.isFastSyncDone(), doFastSync) match {
+        case (false, true) =>
+          self ! StartFastSync
```

Comment:
We need to sort that out. However, I'd merge this PR and and work on Nicolas' issue in a separate branch.